Pagecontainer: Remove unnecessary $( ":focus" ) fallback on page change

gabrielschulhof opened the pull request:
Fixes gh-1560

arschmitz committed:
Widgets: Pull in new widgets from ui and remove old widgets

This commit pulls in the new jQuery UI Button, Checkboxradio, controlgroup,
and the jQuery UI Accordion. It also Updates to the newest version of
the widget factory which includes the classes option. This commit includes a
backcompat extension for switching boolean style options to classes option.
Lastely it pulls setting theme out into its own extension.

Commit:
Tabs: Override _isLocal() to account for Ajax nav

We cannot compare hrefs to location.href

Fixes gh-7169
Fixes gh-7725
Closes gh-7727

Commit:
Widgets: Updates based on new UI widgets

jaspermdegroot committed:
All: Updated button and icon classes

Fixes gh-7664

Commit:
Page: Rename option "closeButton" to "closeBtn" in dialog extension test

arschmitz commented on js/widgets/pagecontainer.js:
```diff
@@ -939,7 +939,7 @@ define( [
 				alreadyThere, newPageTitle,
 				params,	cssTransitionDeferred,
-				beforeTransition;
+				beforeTransition, focusElement;
```
Follow style guide for variable declarations

gabrielschulhof replied:
I'm not sure what you mean here. Variables that are not being initialized are being listed together at the beginning of the declaration, subsequent lines are indented, the new variable uses complete words. Where does this deviate from the style guide?

arschmitz commented on js/widgets/pagecontainer.js:
```diff
@@ -1036,21 +1036,18 @@ define( [
-			//            handler so we already have the element in hand at this
-			//            point.
-			// Wrap this in a try/catch block since IE9 throw "Unspecified error" if
+			// Wrap this in a try/catch block since IE9 throws "Unspecified error" if
```
Make this a proper support comment

arschmitz commented on js/widgets/popup.js:
```diff
@@ -26,6 +26,17 @@ define( [
 //>>excludeEnd("jqmBuildExclude");
 (function( $, undefined ) {
 
+// Safely handle IE9 fragility
```
Make this a proper support comment

Commit:
Textinput: Implement classes option

Fixes gh-7730

arschmitz commented:
@gabrielschulhof going to put a temp hold on this, because we are considering adding `$.ui.safeActiveElement()` to ui core as well as a form of safe blur ( though the api's will likely be a bit different ).

Commit:
Navigation: Avoid redirect error by redirecting with plain JS

arschmitz committed:
Tests: Update to improve ouput and run speed

Simplifys the runs to just be one per jquery version builds on all branches

arschmitz committed:
Tests: Update to improve ouput and run speed

Simplify travis runs to just be one per jquery version

jaspermdegroot committed:
Icons: Switch to inline-block

jaspermdegroot committed:
Demos: First pass on updating to inline-block icons

arschmitz committed:
Build: Split up grunt file into individual modules

Also removes some old unused tasks and does a lot of general clean up
including changing most of the task aliases. In addition this splits
up travis runs and only builds and test demos once.

Closes gh-8058

arschmitz committed:
Build: Add cache directory to .travis.yml

arschmitz committed:
Build: Make excludes work with individual files

Also add some excludes to travis build to deal with travis only failures

arschmitz committed:
Build: Style fixes to comply with JSCS and new style guide

arschmitz committed:
Accordion: Style fixes for jscs

arschmitz committed:
Controlgroup: Style fixes for jscs

arschmitz committed:
Button: Style fixes for jscs

arschmitz committed:
Checkboxradio: Style fixes for jscs

arschmitz committed:
Tabs: Style fixes for jscs

arschmitz committed:
Widget: Style fixes for jscs

Commit:
All: Update boilerplate

Fixes gh-7973
Closes gh-8062

arschmitz committed:
Build: Add esformatter and format build files

Closes gh-8079

arschmitz committed:
Zoom: Style fixes powered by esformatter

Closes gh-8084

arschmitz committed:
Transitions: Style fixes powered by esformatter

Closes gh-8086

arschmitz committed:
Toolbar: Style fixes powered by esformatter

Closes gh-8085

arschmitz committed:
Textinput: Style fixes powered by esformatter

Closes gh-8113

arschmitz committed:
Tabs: Style fixes powered by esformatter

Closes gh-8112

arschmitz committed:
Table: Style fixes powered by esformatter

Closes gh-8111

arschmitz committed:
Support: Style fixes powered by esformatter

Closes gh-8110

arschmitz committed:
Slider: Style fixes powered by esformatter

Closes gh-8109

arschmitz committed:
Select: Style fixes powered by esformatter

Closes gh-8108

arschmitz committed:
Popup: Style fixes powered by esformatter

Closes gh-8107

arschmitz committed:
Path: Style fixes powered by esformatter

Closes gh-8106

arschmitz committed:
Panel: Style fixes powered by esformatter

Closes gh-8105

arschmitz committed:
Page: Style fixes powered by esformatter

Closes gh-8104

arschmitz committed:
Navbar: Style fixes powered by esformatter

Closes gh-8103

arschmitz committed:
Init: Style fixes powered by esformatter

Closes gh-8102

arschmitz committed:
Flipswitch: Style fixes powered by esformatter

Closes gh-8101

arschmitz committed:
Loader: Style fixes powered by esformatter

Closes gh-8100

arschmitz committed:
Helpers: Style fixes powered by esformatter

Closes gh-8099

arschmitz committed:
Filterable: Style fixes powered by esformatter

Closes gh-8098

arschmitz committed:
Navigation: Style fixes powered by esformatter

Closes gh-8097

arschmitz committed:
Dialog: Style fixes powered by esformatter

Closes gh-8096

arschmitz committed:
Core: Style fixes powered by esformatter

Closes gh-8095

arschmitz committed:
button: Style fixes powered by esformatter

Closes gh-8094

arschmitz committed:
collapsible: Style fixes powered by esformatter

Closes gh-8093

arschmitz committed:
listview: Style fixes powered by esformatter

Closes gh-8092

arschmitz committed:
Toolbar: Style fixes for fixed extension powered by esformatter

Closes gh-8091

arschmitz committed:
degradeInputs: Style fixes powered by esformatter

Closes gh-8090

arschmitz committed:
Events: Style fixes powered by esformatter

Closes gh-8089

arschmitz committed:
clearButton: Style fixes powered by esformatter

Closes gh-8087

arschmitz committed:
vmouse: Style fixes powered by esformatter

Closes gh-8083

arschmitz committed:
tests: Style fixes powered by esformatter

arschmitz committed:
Widget: Style fixes powered by esformatter

Closes gh-8082

arschmitz committed:
Core: Style fixes powered by esformatter

Closes gh-8080

cgack committed:
Rangeslider: Implement Classes Option

Fixes gh-7717
Closes gh-8116

cgack committed:
Flipswitch: Implement Classes Option

Fixes gh-7715
Closes gh-8118

Commit:
Widget: Appease jscs in backcompat

arschmitz commented:
@gabrielschulhof if you base this off 1.5-dev we are on up-to-date ui which contains these now

arschmitz committed:
Page: change option enhance to enhanceWithin

We cant use the name enhance because it potentially conflicts with
data-enhance=false for ignoreContentEnabled

arschmitz committed:
DegradeInputs: Remove code supporting an IE8 bug that is no longer

reproducible with current code.

arschmitz committed:
Build: add more files to jscs:good

arschmitz committed:
Button: Fix incorrect method name

arschmitz committed:
Widget: make widget.js just define $.mobile.widget

arschmitz committed:
Page: use new enhancer module

arschmitz committed:
Popup: Use new enhancer module to do link modifications

arschmitz committed:
Build: Remove links.js and nojs.js

arschmitz committed:
Widget: remove mobile getCreateOptions extension

This is now part of the enhancer module also removes tests for this
and the page based enhancer

arschmitz committed:
Collapsibleset: remove deprecated collapsible-set name

arschmitz committed:
Fieldcontain: Remove deprecated fieldcontain feature

arschmitz committed:
Fieldcontain: Remove deprecated fieldcontain feature

arschmitz committed:
Dialog: Remove tests for deprecated data-role=content

arschmitz committed:
Listview: update use enhancer module

arschmitz committed:
Page: No longer handle links.js

This is moved from the central enhancer and page to popup itself

arschmitz committed:
Navbar: use new enhancer module in tests

arschmitz committed:
Texinput: fix bad option name

caused by bad search and replace which changed btn to button

arschmitz committed:
Demos: Use new enhancer module

arschmitz committed:
Selectmenu: Stop using deprecated page option for keepNative

arschmitz committed:
Selectmenu: Update tests to use new enhancer module

arschmitz committed:
Slider: Stop using deprecated page option for keepNative

arschmitz committed:
Slider: Use new enhancer module in tests

arschmitz committed:
Enhancer: Add new enhancer module

arschmitz committed:
DegradeInput: Update to use new enhancer module

Also adds tests for degrade inputs

arschmitz committed:
DegradeInput: Update to use new enhancer module

Also adds tests for degrade inputs

arschmitz committed:
Toolbar: use new enhancer module

arschmitz committed:
Dialog: update to use new pagecontainer method for background

arschmitz committed:
Page: remove old page based enhancer

arschmitz committed:
Widget: Fixup extend backcompat and theme extensions

Commit:
All: Make sure events are well-formed

Fixes gh-8134
Closes gh-8135

arschmitz committed:
Widget: Fixup theme extension to allow specificying of the option to use

jaspermdegroot committed:
Textinput: Update CSS

jaspermdegroot committed:
Textinput: Removed the space after the search icon

arschmitz committed:
Build: remove fieldcontain.js from index.php it no longer exists

arschmitz committed:
Init: Switch .load with .on the former was deprecated and removed

arschmitz committed:
Navigation: Replace .load with .on the former was deprecated and removed

arschmitz committed:
Toolbar: Update fixedToolbar backcompat to new UMD wrapper

arschmitz committed:
Build: Update dependencies

arschmitz commented:
Can you also land this on 1.5-dev please

arschmitz committed:
Widget: Upadtes for theme and backcompat

arschmitz committed:
Button: Fix backcompat dependencies

Commit:
Pagecontainer: Remove unnecessary $( ":focus" ) fallback on page change

Fixes gh-1560

gabrielschulhof commented:
Reopened as https://github.com/jquery/jquery-mobile/pull/8154 against 1.5-dev.
